fix(core): Make password-reset urls valid only for single-use

netroy opened the pull request.

netroy committed:
fix(core): Make password-reset urls valid only for single-use

RicardoE105 commented:
@netroy On dev mode for some reason the env variable `config.getEnv('userManagement.jwtSecret');` is not set, and makes the signing of the token fail with `Error: secretOrPrivateKey must have a value`. Looking into it

RicardoE105 commented:
Bellow what we get when clicking on the reset password link after changing the password. Should we redirect to the forgot password page and tell them that the link was already use? 

<img width="1920" alt="image" src="https://github.com/n8n-io/n8n/assets/16496553/3ef10582-f854-4400-97f7-5e92e90c0f66">

RicardoE105 reviewed (dismissed):
LGTM! left one small comment but I guess that can be addressed in another PR.

netroy commented on packages/cli/src/services/jwt.service.ts:
```diff
@@ -4,14 +4,12 @@ import config from '@/config';
-
 	public signData(payload: object, options: jwt.SignOptions = {}): string {
-		return jwt.sign(payload, this.userManagementSecret, options);
+		return jwt.sign(payload, config.getEnv('userManagement.jwtSecret'), options);
```
this had to made because of a bug in dev mode that [this](https://github.com/n8n-io/n8n/pull/7550) PR fixes

netroy committed:
on invalid password-reset token, show error, and redirect to the login screen

netroy commented:
reverted the `config.getEnv` change, as that should be fixed in #7550.

Updated the frontend to show the correct error, and the also redirect to the signin screen, just how we do it when the invitation token in invalid on the signup screen.

janober commented:
Got released with [`n8n@1.16.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.16.0)
